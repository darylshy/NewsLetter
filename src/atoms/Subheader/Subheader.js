import React from 'react';
import PropTypes from 'prop-types';
import classes from './Subheader.module.scss';

const Subheader = props => {
  return <h3 className={classes.Subheader}>{props.subheader}</h3>;
};
Subheader.defaultProps = {
  subheader: 'Lorem, ipsum dolor sit amet consectetur.'
};
Subheader.propTypes = {
  subheader: PropTypes.string.isRequired
};

export default Subheader;
