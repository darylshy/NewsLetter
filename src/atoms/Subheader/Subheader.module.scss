@import "../../styles/global/globals.scss";
.Subheader {
  font-family: "Azo-Sans-Light";
  margin: 0;
  @include desktop {
    font-size: 2.6rem;
  }
  @include tabletL {
    font-size: 2rem;
  }
  @include tabletP {
    font-size: 2rem;
  }
  @include mobile {
    font-size: 1.8rem;
  }
}
