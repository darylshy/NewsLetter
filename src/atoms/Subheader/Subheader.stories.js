import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text } from '@storybook/addon-knobs/react';
import React from 'react';

import Subheader from './Subheader';

storiesOf('Atoms/Subheader', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add(
    'Standard Subheader Element',
    () => (
      <div style={{ color: '#94aebf' }}>
        <Subheader subheader={text('subheader')} />
      </div>
    ),
    {
      info: {
        text: 'Subheader Element'
      }
    }
  );
