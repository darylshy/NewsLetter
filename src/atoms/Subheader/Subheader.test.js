import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Subheader from './Subheader';
import React from 'react';

configure({ adapter: new Adapter() });

describe('<Subheader />', () => {
  let component;
  beforeEach(() => {
    component = shallow(<Subheader subheader="subheader" />);
  });
  it('should render an Subheader component', () => {
    expect(component).toMatchSnapshot();
  });
  it('should find a Subheader', () => {
    expect(component.find('h3')).toHaveLength(1);
  });
});
