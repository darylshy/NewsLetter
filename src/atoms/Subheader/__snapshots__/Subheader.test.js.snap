// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subheader /> should render an Subheader component 1`] = `ShallowWrapper {}`;
